### PR TITLE
Add documentation for installing plugins in air-gapped environments

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Add documentation for installing plugins in air-gapped environments

### Which issue(s) this PR fixes

Related to #223

### Release note

```release-note
NONE
```

### CODE REVIEW

File: docs/full/README.md [hunk 1]
```diff
@@ -206,3 +206,83 @@ Signature verification could fail in the scenarios below:
...
+However, If you want to just migrate plugins within specific plugin groups
+(e.g. `vmware-tkg/default:v2.1.0`) you can run the below command to download
+the plugin bundle containing only plugins from specified groups:
+
+```sh
+tanzu plugin download-bundle --group vmware-tkg/default:v2.1.0 --to-tar /tmp/plugin_bundle_tkg_v2_1_0.tar.gz
+```
+
+To migrate plugins from the specific plugin repository and not use the default
+plugin repository you can provide a `--image` tag with the above command. Example:
```
Comment (marckhouzam) [inline docs/full/README.md:249]:
```suggestion
plugin repository you can provide a `--image` flag with the above command. Example:
```

File: docs/full/README.md [hunk 2]
```diff
@@ -206,3 +206,83 @@ Signature verification could fail in the scenarios below:
...
+
+Once you download the plugin bundle as a `tar.gz` file and copy the file to the
+air-gapped network, you can run the following command to migrate plugins to the
+private registry (e.g. `registry.example.com/tanzu-cli/plugin`).
+
+```sh
+tanzu plugin upload-bundle --tar /tmp/plugin_bundle_complete.tar.gz --to-repo `registry.example.com/tanzu-cli/plugin`
+```
+
+The above-mentioned commands uploads the plugin bundle to the provided private
```
Comment (marckhouzam) [inline docs/full/README.md:268]:
```suggestion
The above-mentioned command uploads the plugin bundle to the provided private
```

File: docs/full/README.md [hunk 3]
```diff
@@ -206,3 +206,83 @@ Signature verification could fail in the scenarios below:
...
+repository location with the image name `plugin-inventory:latest`. So for the
+above example, the plugin inventory image will be published to
+`registry.example.com/tanzu-cli/plugin/plugin-inventory:latest`.
+
+Please note that `tanzu plugin upload-bundle` uploads the plugins by merging
```
Comment (marckhouzam) [inline docs/full/README.md:273]:
```suggestion
Please note that `tanzu plugin upload-bundle` uploads the plugins by adding them
```

File: docs/full/README.md [hunk 4]
```diff
@@ -206,3 +206,83 @@ Signature verification could fail in the scenarios below:
...
+the plugin-inventory if exists already. That means if you have already uploaded
```
Comment (marckhouzam) [inline docs/full/README.md:274]:
```suggestion
to any plugin-inventory already present in the private registry. That means if you have already uploaded
```

File: docs/full/README.md [hunk 5]
```diff
@@ -206,3 +206,83 @@ Signature verification could fail in the scenarios below:
...
+plugins and append new plugins from the plugin bundle provided.
+
+You can use this image and configure the default discovery source to point to
+this image by running the following command:
+
+```sh
+tanzu plugin source update default --uri registry.example.com/tanzu-cli/plugin/plugin-inventory:latest
+```
+
+Now, Tanzu CLI should be able to discover plugins from this newly configured
```
Comment (marckhouzam) [inline docs/full/README.md:285]:
```suggestion
Now, the Tanzu CLI should be able to discover plugins from this newly configured
```

**Review by marckhouzam — COMMENTED** [review]
This is great!  The comments are mostly grammatical.

For each section, could we add a first line saying "The below steps should be performed by the operator of the private registry" or "The below steps should be performed by users of the Tanzu CLI"?

Could we also put a note to "Refer to ..." @prkalle section on how to setup certificates for air-gapped from #252.

**Review by marckhouzam — COMMENTED** [review]
Should we also have a section to quickly mention how to install the CLI itself in an air-gapped env?

**Comment by anujc25** [comment]
> Should we also have a section to quickly mention how to install the CLI itself in an air-gapped env?

Added a small section at the top.

**Review by marckhouzam — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**